api_fhir_r4/converters/claimConverter.py

```diff
@@ -26,13 +28,22 @@ def to_fhir_obj(cls, imis_claim):
         fhir_claim = FHIRClaim()
         cls.build_fhir_pk(fhir_claim, imis_claim.uuid)
         fhir_claim.created = imis_claim.date_claimed.isoformat()
-        fhir_claim.facility = HealthcareServiceConverter.build_fhir_resource_reference(imis_claim.health_facility)
+        if imis_claim.health_facility is None:
+            raise FHIRRequestProcessException([F'Cannot construct a {imis_claim.uuid} claim if HF is None'])
+        fhir_claim.facility = HealthcareServiceConverter.build_fhir_resource_reference(imis_claim.health_facility,
+                                                                                       imis_claim.health_facility.code)
         cls.build_fhir_identifiers(fhir_claim, imis_claim)
-        fhir_claim.patient = PatientConverter.build_fhir_resource_reference(imis_claim.insuree)
+        if imis_claim.insuree is None:
+            raise FHIRRequestProcessException([F'Cannot construct a {imis_claim.uuid} claim if Insuree is None'])
+        fhir_claim.patient = PatientConverter.build_fhir_resource_reference(imis_claim.insuree,
+                                                                            imis_claim.insuree.chf_id)
         cls.build_fhir_billable_period(fhir_claim, imis_claim)
         cls.build_fhir_diagnoses(fhir_claim, imis_claim)
         cls.build_fhir_total(fhir_claim, imis_claim)
-        fhir_claim.enterer = PractitionerConverter.build_fhir_resource_reference(imis_claim.admin)
+        if imis_claim.admin is not None:
+            fhir_claim.enterer = PractitionerConverter.build_fhir_resource_reference(imis_claim.admin, 'Practitioner')
+        else:
+            raise FHIRRequestProcessException([F'Cannot construct a {imis_claim.uuid} claim if claim admin is None'])
```

Comment:
Exception message strange: 'Cannot construct a UUID claim if claim admin is None'. Need to be updated.

Reply:
This exception was removed in master branch later on so I removed it as well. 
It is different from HF and insuree because the value in the database may be null, in that case should we allow no admin in imis -> fhir mapping or is it better to keep this exception?

Reply:
The ClaimAdmin is a mandatory field. We need to keep the exception.

Reply:
I'll restore it, how we should change exception message? 
Will change to `'Failed to create FHIR instance for claim {imis_claim.uuid}: Claim Admin field not found'` be adequate?

Reply:
You can change to that message. We will have to enforce the ClaimAdmin field in the DB so in the future there will be no exception.

Reply:
Exception restored